### PR TITLE
Reduce EVM module dependencies

### PR DESCRIPTION
**Pull request by shemnon**
## PR description

Refactor crypto, datatypes, and plugin-api to reduce plugin APIs in the EVM.

Split crypto into crypto services and crypto algorithms
Reverse the dependency between datatyps and plugin-api.
Remove plugin Hash and Address types (use datatypes)
Move PublicKey and Quantity into datatypes.

Lots of changes to imports and build files, and some `fromPlugin` calls removed.

### CODE REVIEW

File: crypto/algorithms/build.gradle [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright ConsenSys AG.
```
Comment (jflo) [inline crypto/algorithms/build.gradle:2]:
```suggestion
 * Copyright Hyperledger Besu Contributors
```

File: crypto/services/build.gradle [hunk 2]
```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright ConsenSys AG.
```
Comment (jflo) [inline crypto/services/build.gradle:2]:
```suggestion
 * Copyright Hyperledger Besu Contributors
```

File: ethereum/core/src/test/java/org/hyperledger/besu/ethereum/privacy/PrivateStateGenesisAllocatorTest.java [hunk 3]
```diff
@@ -56,7 +56,7 @@ public class PrivateStateGenesisAllocatorTest {
           List.of(
               new PrivacyGenesisAccount() {
                 @Override
-                public org.hyperledger.besu.plugin.data.Address getAddress() {
+                public org.hyperledger.besu.datatypes.Address getAddress() {
```
Comment (jflo) [inline ethereum/core/src/test/java/org/hyperledger/besu/ethereum/privacy/PrivateStateGenesisAllocatorTest.java:59]: Might not need to be fully qualified here.

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiWorldStateProvider.java [hunk 4]
```diff
@@ -337,7 +335,7 @@ public TrieLogManager getTrieLogManager() {
 
   @Override
   public void resetArchiveStateTo(final BlockHeader blockHeader) {
-    persistedState.resetWorldStateTo(blockHeader);
+    persistedState.setArchiveStateUnSafe(blockHeader);
```
Comment (jflo) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiWorldStateProvider.java:338]: This seems out of place, merge/rebase issue?

**Review by jflo — CHANGES_REQUESTED** [review]
BonsaiWorldStateProvider looks like a renaming snuck into this PR from somewhere else?

File: ethereum/core/src/test/java/org/hyperledger/besu/ethereum/privacy/PrivateStateGenesisAllocatorTest.java (hunk 3 above)
Comment (shemnon) [reply ethereum/core/src/test/java/org/hyperledger/besu/ethereum/privacy/PrivateStateGenesisAllocatorTest.java]: fixed

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiWorldStateProvider.java (hunk 4 above)
Comment (shemnon) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiWorldStateProvider.java]: rebase issue.  It was next to some `fromPlugin` removals.

**Review by jflo — APPROVED** [review]

**Review by Gabriel-Trintinalia — APPROVED** [review]

**Pull request merged**